modules/misunderstood/src/backend/db.ts

```diff
@@ -103,6 +106,75 @@ export default class Db {
     }))
   }
 
+  async importData(botId: string, events: FlaggedEvent[]): Promise<void> {
+    const botGhost = this.ghostForBot(botId)
+
+    // Only deal with events that have a resolution defined. They can be null and pass the type check
+    // because the type allows for un-resolved events, but we can't import those here
+    events = events.filter(d => d.resolution && d.resolutionType)
+    if (events.length === 0) {
+      return
+    }
+
+    // sort intents and QnAs
+    const [qnaEvents, nluEvents] = _.partition(events, { resolutionType: RESOLUTION_TYPE.qna })
+
+    // Assert the intents and QnAs already exist
+    const fileList = [
+      ...(await botGhost.directoryListing('intents', '*.json')),
+      ...(await botGhost.directoryListing('qna', '*.json'))
+    ]
+    let missing: Array<string> = []
```

Comment:
You don't reassign missing, and please use string[] instead of Array<string>

`const missing: string[] = []`